plugins/org.jkiss.dbeaver.erd.ui/src/org/jkiss/dbeaver/erd/ui/notation/crowsfoot/CrowsFootDiagramNotation.java

```diff
@@ -0,0 +1,86 @@
+import org.jkiss.dbeaver.model.struct.DBSEntityConstraintType;
+
+public class CrowsFootDiagramNotation extends ERDNotationBase implements ERDNotation {
+    @Override
+    public void applyNotationForArrows(PolylineConnection conn, ERDAssociation association, Color bckColor, Color frgColor) {
+        DBSEntityConstraintType constraintType = association.getObject().getConstraintType();
+        if (constraintType == DBSEntityConstraintType.PRIMARY_KEY) {
+            // source 0..1
+            final CrowsFootPolylineDecoration sourceDecor = new CrowsFootPolylineDecoration(ERDAssociationType.ZERO_OR_ONE);
+            sourceDecor.setFill(true);
+            sourceDecor.setBackgroundColor(bckColor);
+            conn.setSourceDecoration(sourceDecor);
+        } else if (constraintType.isAssociation() &&
+            association.getSourceEntity() instanceof ERDEntity &&
+            association.getTargetEntity() instanceof ERDEntity) {
+            // source - 1..n
+            final CrowsFootPolylineDecoration sourceDecor = new CrowsFootPolylineDecoration(ERDAssociationType.ONE_OR_MANY);
+            sourceDecor.setFill(true);
+            sourceDecor.setBackgroundColor(bckColor);
+            conn.setSourceDecoration(sourceDecor);
+            ConnectionEndpointLocator srcEndpointLocator = new ConnectionEndpointLocator(conn, false);
+            srcEndpointLocator.setVDistance(-10);
+            srcEndpointLocator.setUDistance(5);
+            Label label = new Label(LABEL_1_TO_N);
+            conn.add(label, srcEndpointLocator);
+            if (ERDUtils.isOptionalAssociation(association)) {
+                // target - 0..1
+                final CrowsFootPolylineDecoration targetDecor = new CrowsFootPolylineDecoration(ERDAssociationType.ZERO_OR_ONE);
+                targetDecor.setFill(true);
+                targetDecor.setBackgroundColor(bckColor);
+                ConnectionEndpointLocator trgEndpointLocator = new ConnectionEndpointLocator(conn, true);
+                trgEndpointLocator.setVDistance(-10);
+                trgEndpointLocator.setUDistance(5);
+                conn.add(new Label(LABEL_0_TO_1), trgEndpointLocator);
+                conn.setTargetDecoration(targetDecor);
+            } else {
+                // target - 1
+                final CrowsFootPolylineDecoration targetDecor = new CrowsFootPolylineDecoration(ERDAssociationType.ONE_ONLY);
+                targetDecor.setFill(true);
+                targetDecor.setBackgroundColor(bckColor);
+                ConnectionEndpointLocator trgEndpointLocator = new ConnectionEndpointLocator(conn, true);
+                trgEndpointLocator.setVDistance(-5);
+                trgEndpointLocator.setUDistance(10);
+                conn.add(new Label(LABEL_1), trgEndpointLocator);
+                conn.setTargetDecoration(targetDecor);
+            }
+        }
+        conn.setLineWidth(1);
+        conn.setLineStyle(SWT.LINE_CUSTOM);
```

Comment:
Also, provide screenshots with the shortest path and orthogonal routing.

Reply:
![image](https://github.com/dbeaver/dbeaver/assets/32031248/60912837-61b0-48c4-9cdf-13a8c11e480d)